resource-manager/src/main/java/org/ihtsdo/otf/resourcemanager/ResourceManager.java

```diff
@@ -1,113 +1,368 @@
+	 * @return an {@code InputStream} which corresponds to the resource
+	 * stream.
+	 * @throws IOException If an error occurs while trying to load
+	 *                     the resource.
+	 */
+	public InputStream readResourceStream(final String resourcePath) throws IOException {
+		final String fullPath = getFullPath(resourcePath);
 		try {
-			Resource resource = resourceLoader.getResource(fullPath);
-			return resource.getInputStream();
+			return resourceLoader.getResource(fullPath).getInputStream();
 		} catch (AmazonS3Exception e) {
 			throw new IOException("Failed to load resource '" + fullPath + "'.", e);
 		}
 	}
 
-	public InputStream readResourceStreamOrNullIfNotExists(String resourcePath) throws IOException {
+	/**
+	 * Reads the resource from the given <code>resourcePath</code>. It will
+	 * return {@code null} if the {@code Resource} does not exist.
+	 *
+	 * @param resourcePath Path to the resource.
+	 * @return an {@code InputStream} which corresponds to the resource stream.
+	 * Otherwise, it will return {@code null} if the {@code Resource} does not
+	 * exist.
+	 * @throws IOException If an error occurs while trying to load the resource.
+	 */
+	public InputStream readResourceStreamOrNullIfNotExists(final String resourcePath) throws IOException {
 		try {
-			String fullPath = getFullPath(resourcePath);
-			Resource resource = resourceLoader.getResource(fullPath);
+			final Resource resource = resourceLoader.getResource(getFullPath(resourcePath));
 			return resource.exists() ? resource.getInputStream() : null;
 		} catch (AmazonS3Exception e) {
 			throw new IOException("Failed to load resource '" + resourcePath + "'.", e);
 		}
 	}
 
-	public void writeResource(String resourcePath, InputStream resourceInputStream) throws IOException {
+	/**
+	 * Writes the data to the given resource path location.
+	 *
+	 * @param resourcePath        Which points to the resource that is
+	 *                            going to have the write operation performed
+	 *                            on it.
+	 * @param resourceInputStream Input stream which represents the converted
+	 *                            object, encoded as bytes.
+	 * @throws IOException If an error occurs while trying to write the
+	 *                     input stream to the specified resource path.
+	 */
+	public void writeResource(final String resourcePath,
+							  final InputStream resourceInputStream) throws IOException {
 		try {
-			try (OutputStream outputStream = writeResourceStream(resourcePath);
-				 InputStream inputStream = resourceInputStream) {
+			try (final OutputStream outputStream = retrieveWritableResourceStream(resourcePath);
+				 final InputStream inputStream = resourceInputStream) {
 				StreamUtils.copy(inputStream, outputStream);
 			}
 		} catch (AmazonS3Exception e) {
 			throw new IOException("Failed to write resource '" + resourcePath + "'.", e);
 		}
 	}
 
-	public OutputStream writeResourceStream(String resourcePath) throws IOException {
```

Comment:
This method is used in the [Classification Service ClassificationJobManager](https://github.com/IHTSDO/classification-service/blob/4.3.0/src/main/java/org/snomed/otf/reasoner/server/service/ClassificationJobManager.java#L149). I think we still need it.

Reply:
Right I see, thanks. I will make it public again but keep the name as it I've done it already as `writeResourceStream` doesn't make sense for what the underlying functionality of it does.

Reply:
yeah, it's more like open resource output stream 👍